### DISCUSSION TITLE
Transfer attributes from parent to child feature using `@current_parent_feature`

### DISCUSSION THREAD

**Discussion by cratcliff** [opengisch/QField#5327] [Q&A]
I'm trying to transfer values from a parent feature to a child by using `attribute(@current_parent_feature,'Site')` as the default expression and then use the values for photo filenames. 

I am trying to use `@current_parent_feature`  over other solutions like `attribute(get_attribute(  )   )` as the layer name/id and relationship name/ids will change.

Currently the values from the parent are _retrieved and displayed_ in child form correctly, but _aren't saved or accessible_ unless edited (i.e. add a space/hyphen). As a result they aren't accessible by the photo naming expression either.

Is it possible to use `@current_parent_feature` in this way with Qfield?

The following shows the behavior I am experiencing.
<a href="url"><img src="https://github.com/opengisch/QField/assets/29470818/5f9e1a70-56d9-497e-b869-f0738876a367" height="350" ></a>

I have also included a simplified [QGIS project](https://github.com/user-attachments/files/15781892/QGIS_PhotosCollection.zip)  &  [packaged QField project](https://github.com/user-attachments/files/15781894/QField_PhotosCollection.zip) to demonstrate.

I am using QGIS 3.36.2-Maidenhead for Windows, QField Sync 4.9.1 and Qfield 3.2.2-Congo on Android 14 (Samsung Active 4 Pro).

**Comment by m-kuhn** [comment]
Thank you for reporting this, @cratcliff 

We have made a couple of improvements in this area, could you check if this is still an issue with QField 3.3?

**Reply by cratcliff** [reply]
@m-kuhn Yes, it is still an issue. I'm now using Qfield 3.3.1-Darien(6b9cd v3.3.1) and QGIS 3.36.3-Maidenhead.

**Reply by cratcliff** [reply]
@m-kuhn I've just upgraded again to 3.3.2 Darien (a2b463 v3.3.2) and it is still an issue.

**Comment by nirvn** [comment]
@cratcliff , thanks for providing a test project. A fix to the issue raised in this discussion has been found. You can test the APKs that will appear here to confirm it works as expected: https://github.com/opengisch/QField/pull/5356

**Reply by 9ls1** [reply]
`@current_parent_feature` doesn't seem to work in 3.7.9? See [https://github.com/opengisch/QField/issues/6862](https://github.com/opengisch/QField/issues/6862)

**Comment by 9ls1** [comment]
@cratcliff, did you test the APK and can you confirm that `@current_parent_feature` works in the photo naming expression? I can't get it working in QField version 3.7.9.

**Comment by 9ls1** [comment]
Problem solved, thanks to @FelixHinckel, see https://community.qfield.org/t/using-current-parent-feature-in-filename-expression-not-working/1379

**Comment by Kalana-maduranga001** [comment]
Is @current_parent_feature expected to work in photo naming expressions in QField, or is it currently limited to form display only?

**Comment by Kalana-maduranga001** [comment]
When using attribute(@current_parent_feature, 'field') as a default value in a child layer, should the value be persisted automatically on save, or is manual editing required for it to be stored?

**Comment by FelixHinckel** [comment]
For photo naming, you can use the fonction, rater than the variable : 
`current_parent_value( 'Field' )` works very well for me

I use it as a default value for a field in the child layer, and then use that field in photo naming

**Reply by Kalana-maduranga001** [reply]
Thanks for the suggestion 

Using the **function** instead of the variable solved the issue for me.

For photo naming, `@current_parent_feature` does not seem to be reliably resolved, but using:

current_parent_value('Field')
works correctly.

I now use `current_parent_value('Field')` as the **default value** for a field in the child layer, and then reference that **stored child field** in the photo naming expression. With this approach, the value is properly saved and consistently available when the photo is created.

This seems to be the recommended workaround for photo naming when parent attributes are required, without relying on layer or relationship IDs.

Hope this helps others running into the same issue.
